src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -634,17 +614,20 @@ LookupTaskHandle PosixEventEngine::PosixDNSResolver::LookupTXT(
           // on_resolve called, no longer inflight.
           GPR_ASSERT(inflight_requests_.erase(handle) == 1);
         }
+        reinterpret_cast<GrpcAresRequest*>(handle.keys[0])->Orphan();
         on_resolve(std::move(result));
       });
   return handle;
 }
 
 bool PosixEventEngine::PosixDNSResolver::CancelLookup(LookupTaskHandle handle) {
-  grpc_core::MutexLock lock(&mu_);
+  grpc_core::ReleasableMutexLock lock(&mu_);
```

Comment:
Cool, thanks for the suggestion! This part is not needed in the new implementation since we don't have `CancelLookup` method anymore. (responding ahead of time since the next update will make this comment outdated)